ReviewChecklist.md

```diff
@@ -0,0 +1,166 @@
+common issues before they are inevitably pointed out in the review.
+It's also meant as a final checklist for reviewers to go through before approving a PR.
+
+## Before you submit a PR
+- [ ] **Is the issue ready to be worked on?**
+    - If the issue does not have a desirability label (`nice to have`, `should have`,
+        `must have eventually`, `must have`, `roadmap`) we still have not decided whether to implement it.
+    - If the issue `needs design`, we have not yet decided how it should be implemented.
+    - `good first issue candidate` means that the issue will potentially be a `good first issue`
+        eventually but at the moment it's not yet ready to be worked on.
+- [ ] **Is this a breaking change?** Breaking changes should be based on the `breaking` branch rather than on `develop`.
+- [ ] **Does the PR actually address the issue?**
+    - [ ] Mention the issue number in the PR description.
+        If the PR solves it completely, use the `Fixes #<issue number>` form so that Github will close the issue automatically
+    - [ ] Do not include issue number in the PR title, branch name or commit description.
+- [ ] When submitting a PR from a fork **create a branch and give it a relevant name.**
+    E.g. `fix-array-abi-encoding-bug`.
+    Do not submit PRs directly from the `develop` branch of a fork since it makes rebasing and fetching new changes harder.
+- [ ] **Does the PR depend on other PRs?**
+    - [ ] If the PR has dependencies, mention them in bold in the description.
+    - [ ] Avoid basing PRs from forks on branches other than `develop` or `breaking` because
+        GitHub closes them when the base branch gets merged.
+        Do this only for PRs created directly in the main repo.
+
+## Coding style and good practices
+- [ ] Does the PR follow our [coding style](CODING_STYLE.md)?
+
+### Reliability
+- [ ] **Use assertions liberally.** If you are certain your assumption will not be be broken, prove it with `solAssert()`.
+- [ ] **Validate inputs and handle errors**. Note that assertions are **not** validation.
+
+### Readability
+- [ ] **Choose good names.**
+    - [ ] Is the name straightforward to understand?
+        Do you feel the need to jump back to the definition and remind yourself what it was whenever you see it?
+    - [ ] Is the name unambiguous in the context where it's used?
+    - [ ] Avoid abbreviations.
+- [ ] **Source files, classes and public functions should have docstrings.**
+- [ ] **Avoid code duplication.** But not fanatically. Minimal amounts of duplication are acceptable if it aids readability.
+- [ ] **Don't leave dead or commented-out code behind.** You can still see old code in history.
+      But if you really have a good reason to do it, always leave a comment explaining what it is and why it's there.
+- [ ] **Mark hacks as such.** If you have to leave behind something ugly, make sure to include a
```

Comment:
```suggestion
- [ ] **Mark hacks as such.** If you have to leave behind something in a suboptimal state, make sure to include a
```

Reply:
or is it common to refer to something as "ugly"?

Reply:
I think it is, at least casually. "an ugly hack" and such.